src/CommandParser.h

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-2-Clause
```

Comment:
It seems to me that having `CommandParser` as a class is not really justified. The only shared state is the pointer to the logger. I think that the implementation would be slightly cleaner (and easier to test) by making `CommandParser` into a namespace. The pointer to the logger could still be retained as a shared state by removing it from the header file and replacing it with `static SharedLogger Logger = getLogger();` in the *.cpp-file.

Reply:
Agreed, that was something I was thinking off doing then forgot!

Reply:
I'll see how the next set of changes go and see if they fit with this.